website/src/app/portal/components/data-table/clients/recipients-table-client.tsx

```diff
@@ -21,19 +22,16 @@ export function RecipientsTableClient({
 
 	const [recipientId, setRecipientId] = useState<string | undefined>();
 	const [hasError, setHasError] = useState(false);
-
-	const [readOnly, setReadOnly] = useState(false);
+	const readOnly = rows.some((row) => row.permission === ProgramPermission.readonly);
```

Comment:
I’m not fully convinced this is the best approach.

Instead, we could pass a readonly flag into the tableClient, so we only need to decide edit permissions in one place:
	•	On a program-scoped page, we already know whether the user can edit that specific program.
Since page.tsx calls getAuthenticatedUserOrRedirect, which eventually calls UserService.getCurrentUserInformation, we could extend that method to include the required permission info.
	•	On the monitoring page, the button should only be enabled if the user has edit access to at least one program.